client/src/main/scala/drt/client/components/ShiftsComponent.scala

```diff
@@ -14,13 +14,14 @@ import japgolly.scalajs.react.vdom.html_<^.VdomTagOf
 import org.scalajs.dom.html.Div
 import japgolly.scalajs.react.vdom.html_<^._
 import uk.gov.homeoffice.drt.Shift
+import uk.gov.homeoffice.drt.models.UserPreferences
 import uk.gov.homeoffice.drt.ports.Terminals.Terminal
 import uk.gov.homeoffice.drt.time.SDateLike
 import uk.gov.homeoffice.drt.time.LocalDate
 
 object ShiftsComponent {
 
-  case class Props(terminal: Terminal, portCode: String, router: RouterCtl[Loc])
+  case class Props(terminal: Terminal, portCode: String, userPreferences: UserPreferences, shiftAction: String, router: RouterCtl[Loc])
```

Comment:
It would be good to make `shiftAction` a trait with some case objects to represent each action type